### PR TITLE
Allow splash to be JPG also

### PR DESCRIPTION
**Pull request by yol**
## Description
Load Splash.jpg before Splash.png

## Motivation and Context
To have more fancy splash screens, allow loading from JPG instead of PNG for reasonable file sizes

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by pkerling** [comment]
@mkortstiege This drops `GUIWindowSplash` that you added some years ago. Can you assess whether it's OK to drop it now or do we still need it?

File: tools/android/packaging/.gitignore
Comment (koying) [inline tools/android/packaging/.gitignore]: Have to be capital `Splash.*`, now

File: tools/android/packaging/Makefile.in
Comment (koying) [inline tools/android/packaging/Makefile.in]: `xbmc/res/drawable/`
Same below

File: tools/android/packaging/Makefile.in
Comment (koying) [inline tools/android/packaging/Makefile.in]: Have to be capital `Splash.*`, now

**Comment by koying** [comment]
You also have to change https://github.com/xbmc/xbmc/blob/master/tools/android/packaging/xbmc/res/layout/activity_splash.xml#L14 to a camel case
```
android:src="@drawable/Splash" />
```

**Comment by pkerling** [comment]
@koying Very correct, thanks. I kind of want to change it to the lower-case variant everywhere now, but I guess this would only further complicate the situation.

**Comment by notspiff** [comment]
it shouldn't complicate much. afaik we look for the file using a "case-fuzzing" function.

**Comment by pkerling** [comment]
@notspiff doesn't seem to work sadly. Crashes when I just rename to lowercase.

**Comment by mkortstiege** [comment]
@pkerling I am not entirely sure if the actual "real" window is of any use these days. Back then, we needed it for the master lock and profile handling. We also wanted it for some more advanced progress when doing main app init and database upgrades afaik.

**Comment by koying** [comment]
Looks good now. Will test to be sure ;)

File: xbmc/Util.cpp
Comment (notspiff) [inline xbmc/Util.cpp]: pass this through CSpecialProtocol::TranslatePathConvertCase, or the better but more risky fix which is to fix CSpecialProtocolFile::TranslatePath

**Comment by pkerling** [comment]
@mkortstiege the updates that show text should be handled by `CRenderSystemBase` I guess?

**Comment by mkortstiege** [comment]
Jep, guess so. I can recall recent changes by @fernetmenta

pkerling <notifications@github.com> schrieb am Di., 2. Jan. 2018, 13:04:

**Comment by pkerling** [comment]
Filename changed to lowercase `splash` everywhere (I hope...)

**Comment by koying** [comment]
Would be good to switch to lowercase, as droid does not support drawables with a capital in the name :(
Sorry for the noise...

**Comment by da-anda** [comment]
samfisher is redoing my splash in 3D this very second to make it more awesome, so we might want to wait on his version before we merge it?

**Review by koying — APPROVED** [review]
Screen turns to black during profile changes (not sure how it was before really) but no showstopper

**Comment by MartijnKaijser** [comment]
Same happens on master so nothing new i think.

**Comment by pkerling** [comment]
jenkins build this please

**Review by Memphiz — APPROVED** [review]
iOS changes look ok thx

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by xhaggi** [comment]
We introduced the splash window to stay in background until you logged in to your profile if you had masterlock enabled. https://github.com/xbmc/xbmc/pull/7650 please verify if this is fixed before you remove the window.

**Comment by pkerling** [comment]
@xhaggi I tried some combinations of profile selection and master mode but I can't get to the situation you describe. What exactly do I need to do to have the splash window visible in the background on current master?

**Comment by xhaggi** [comment]
Check the forum thread. Enable masterlock with a password or code, restart Kodi, masterlock should appear, then move the mouse around the screen outside the masterlock dialog.

**Comment by pkerling** [comment]
Ah I missed the ask for master lock code on startup since it's in a different settings page. Oddly enough this also does not work for me on master, i.e. there is no splash image in the background and the dialogs leave strange trails when closing. Can anyone else check?

**Comment by xhaggi** [comment]
With current master I got the following result. The render issue is back :(

screenshot004

Used Xperience1080 because Estuary uses a full size keyboard dialog.
screenshot006

**Comment by pkerling** [comment]
Then what should we do with this PR? It does not make the issue worse, but should I leave in the splash window for now?

**Comment by xhaggi** [comment]
Okay found the commit which re-introduce the rendering issue if master lock is enabled.
https://github.com/xbmc/xbmc/pull/10280/commits/9f6eb4d3dfe35bec36f50e69a3fd81058e4334af#diff-02cbf300ebd15bda2591e96d935b0f87R1506

Remove the condition `(newWindowID != WINDOW_SPLASH)` will fix it.

@pkerling you have to leave the CGUIWindowSplash.

**Comment by MartijnKaijser** [comment]
Perhaps add a code comment why that part is actually there (or some one really fixes itl

**Comment by pkerling** [comment]
@xhaggi Are you sure it's OK to just change the `else if` to `else`? From the commit it sounds like it was done to fix some other issue.

**Comment by xhaggi** [comment]
I'll add a PR for it and ping the right dev.

**Comment by xhaggi** [comment]
> Are you sure it's OK to just change the else if to else? From the commit it sounds like it was done to fix some other issue.

I explain why it's wrong in the PR description.

**Comment by xhaggi** [comment]
> Perhaps add a code comment why that part is actually there

We did that while we introduced the the splash window https://github.com/xbmc/xbmc/blob/master/xbmc/Application.cpp#L1126

**Comment by MartijnKaijser** [comment]
@pkerling can you adjust accordingly now https://github.com/xbmc/xbmc/pull/13294 has been merged?

**Comment by pkerling** [comment]
Sure, the verdict is to leave `CGUIWindowSplash` and the code it duplicates as-is?

File: xbmc/Util.h
Comment (xhaggi) [inline xbmc/Util.h]: could you please add a valid doxy comment

File: xbmc/rendering/RenderSystem.cpp
Comment (xhaggi) [inline xbmc/rendering/RenderSystem.cpp]: What's the reason for the curly braces? why not `std::string splashImage = CUtil::GetSplashPath()`?

**Comment by xhaggi** [comment]
> Sure, the verdict is to leave CGUIWindowSplash and the code it duplicates as-is?

Yep, there is some dupe code (CGUIImage) but nothing else because you moved the path logic to  `CUtil::GetSplashPath()`.

File: xbmc/rendering/RenderSystem.cpp
Comment (pkerling) [reply xbmc/rendering/RenderSystem.cpp]: Matter of taste. I don't really care tbh.

Comment (xhaggi) [reply xbmc/rendering/RenderSystem.cpp]: please be consistent because you use `=` in `IOSExternalTouchController.mm`

Comment (pkerling) [reply xbmc/rendering/RenderSystem.cpp]: Yeah I went for minimal changes there because it was only one line, but in principle you're of course right. I think it would be good to have some of this stuff defined in the coding style guidelines maybe?

File: xbmc/Util.h
Comment (pkerling) [reply xbmc/Util.h]: What are you missing here?

Comment (xhaggi) [reply xbmc/Util.h]:
```c++
/*!
 \brief Return the resolved filesystem ...
 \return the path ....
 */
```

Comment (pkerling) [reply xbmc/Util.h]: Isn't that the same but longer? `///` is picked up by doxygen and used for the `brief`.

Comment (xhaggi) [reply xbmc/Util.h]: sure, but be consistent would be appreciated and you could differ the brief from the return description.

File: xbmc/windows/GUIWindowSplash.cpp
Comment (xhaggi) [inline xbmc/windows/GUIWindowSplash.cpp]: ahh, could you please remove this line .. I think it's a left-over or change the log level to debug.

File: xbmc/rendering/RenderSystem.cpp
Comment (ksooo) [inline xbmc/rendering/RenderSystem.cpp]: I would eliminate `splashImage` variable => `CTextureInfo(CUtil::GetSplashPath())`

Or make the var at least `const`.

File: xbmc/windows/GUIWindowSplash.cpp
Comment (ksooo) [reply xbmc/windows/GUIWindowSplash.cpp]: I would eliminate `splashImage` variable => `CLog::Log(LOGINFO, "load splash image: %s", CUtil::GetSplashPath().c_str());`

Or make the var at least `const`.

Comment (ksooo) [reply xbmc/windows/GUIWindowSplash.cpp]: @xhaggi it was `LOGINFO` before. So, why do you want to change that?

File: xbmc/Util.h
Comment (pkerling) [reply xbmc/Util.h]: `///` is already used all over the place. And honestly I don't know what more I could say in `\return`. It would be just the same information as in the brief but worded differently, which is not helpful.

Comment (xhaggi) [reply xbmc/Util.h]: ok leave it if that is such a big thing.

File: xbmc/windows/GUIWindowSplash.cpp
Comment (xhaggi) [reply xbmc/windows/GUIWindowSplash.cpp]: The logging of the path makes no sense here. It's not logged in `CRenderSystemBase` so why should we do it here? I've introduced it in the initial PR so it should be fine if I say drop it ;)

Comment (ksooo) [reply xbmc/windows/GUIWindowSplash.cpp]: okay, I was just curious.

File: xbmc/rendering/RenderSystem.cpp
Comment (pkerling) [reply xbmc/rendering/RenderSystem.cpp]: sure, why not

File: xbmc/windows/GUIWindowSplash.cpp
Comment (pkerling) [reply xbmc/windows/GUIWindowSplash.cpp]: removed

**Comment by xhaggi** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
Only a test failed on osx.

**Review by xhaggi — APPROVED** [review]
@MartijnKaijser your button

**Comment by xhaggi** [comment]
Thanks @pkerling for your contribution

**Comment by da-anda** [comment]
@MartijnKaijser we could have waited for the reworked splash from samfisher (which is progressing very well). But well, now my borked splash will be featured in one nightly build :)

**Pull request merged**
